asset/encoding.go

```diff
@@ -390,16 +390,48 @@ func TxWitnessDecoder(r io.Reader, val any, buf *[8]byte, _ uint64) error {
 	return tlv.NewTypeForEncodingErr(val, "*wire.TxWitness")
 }
 
-func WitnessEncoder(w io.Writer, val any, buf *[8]byte) error {
+// WitnessEncoderWithType is a wrapper around WitnessEncoder that allows the
+// caller to specify th witness type. It's a higher order function that returns
+// an encoder function.
+func WitnessEncoderWithType(encodeType EncodeType) tlv.Encoder {
+	return func(w io.Writer, val any, buf *[8]byte) error {
+		return witnessEncoder(w, val, buf, encodeType)
+	}
+}
+
+func witnessEncoder(w io.Writer, val any, buf *[8]byte,
+	encodeType ...EncodeType) error {
+
 	if t, ok := val.(*[]Witness); ok {
 		if err := tlv.WriteVarInt(w, uint64(len(*t)), buf); err != nil {
 			return err
 		}
 		for _, assetWitness := range *t {
 			var streamBuf bytes.Buffer
-			if err := assetWitness.Encode(&streamBuf); err != nil {
-				return err
+			switch {
+			case len(encodeType) == 1 &&
```

Comment:
Are we using a variadic function just to not break any existing calls? Since there are only two call sites, shouldn't we make this more explicit instead?

Reply:
Yeah I think that makes sense, actually had a bug here earlier due to missing a default case.